plugins/chrome-ux-report-backend/src/service/Query.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { BigQuery } from '@google-cloud/bigquery'; // used Google's official BigQuery SDK.
+
+function createBigQueryClient(config: Config) {
```

Comment:
i recommend moving this into the class.

```ts
export class Query {
  static fromConfig(config: Config) {
    const projectId = config.getString('chromeUXReport.projectId');
    const keyFilename = config.getString('chromeUXReport.keyPath');
    const client = new BigQuery({ projectId, keyFilename });
    return new Query(client);
  }

  constructor(private readonly client: BigQuery) {}
```

I see you construct the client for every query ... if you have to do that for some reason, you can store a client factory instead of the client itself i guess